File: eecs268/2020fall/12pm/2020.09.04.inheritance-part-2/animal-example/Dog.cpp
//Dog.cpp
#include "Dog.h"

Dog::Dog()
{
	age = 1;
	std::cout << "Dog constructed\n";
}

void Dog::doTrick()
{
	std::cout << "It's doing an amazing trick!\n";
}

void Dog::eat()
{
	std::cout << "Dog eating\n";
	isHungry = false;
}

void Dog::sleep()
{
	std::cout << "Dog sleeping\n";
	isHungry = true;
}

File: eecs268/2026summer/2026.07.21.inheritance/animal-demo/dog.py
#dog.py

from animal import Animal

#     Subclass(Superclass)
class Dog(Animal):
    def __init__(self, age, fur_type):
        #call Animal's init
        super().__init__(age)
        print('Dog init called')
        self.fur_type = fur_type
        
    def do_trick(self):
        print('Dog doing an amazing trick!')

    def sleep(self):
        print('Hrp, hrp, hrp. HRP. Hrp. Hrp. HRP!')
